test/built-ins/Array/prototype/toReversed/frozen-this-value.js

```diff
@@ -0,0 +1,15 @@
+// Copyright (C) 2021 Igalia. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-array.prototype.toReversed
+description: >
+  Array.prototype.toReversed works on frozen objects
+features: [change-array-by-copy]
+---*/
+
+var arr = Object.freeze([0, 1, 2]);
+arr.toReversed();
+
+var arrayLike = Object.freeze({ length: 3, 0: 0, 1: 1, 2: 2 });
+Array.prototype.toReversed.call(arrayLike);
```

Comment:
The test for `Array.prototype.with` includes: 

```js
assert.compareArray(result, [0, 3, 2]);
```

And I think the rest of the methods need this as well. Right now it's only testing that these operations don't throw, but doesn't check the actual result. Let's add those to all of the `frozen-this-value.js` tests in this PR that currently do not have them.